packages/block-library/src/query-total/edit.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useBlockProps,
+	InspectorControls,
+	BlockControls,
+} from '@wordpress/block-editor';
+import {
+	PanelBody,
+	ToolbarGroup,
+	ToolbarDropdownMenu,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { resultsFound, displayingResults } from './icons';
+
+export default function QueryTotalEdit( {
+	attributes,
+	setAttributes,
+	className,
+} ) {
+	const { displayType } = attributes;
+
+	const [ totalResults, setTotalResults ] = useState( 0 );
+	const [ currentPage, setCurrentPage ] = useState( 1 );
+	const [ resultsPerPage, setResultsPerPage ] = useState( 10 );
+
+	// Fetch or calculate the total results, current page, and results per page.
+	useEffect( () => {
+		// Mock values for demonstration. Replace with actual query data.
+		const mockTotalResults = 12;
+		const mockCurrentPage = 1;
+		const mockResultsPerPage = 10;
+
+		setTotalResults( mockTotalResults );
+		setCurrentPage( mockCurrentPage );
+		setResultsPerPage( mockResultsPerPage );
+	}, [] );
+
+	// Helper to calculate the range for "Displaying X – Y of Z".
+	const calculateRange = () => {
+		const start = ( currentPage - 1 ) * resultsPerPage + 1;
+		const end = Math.min( currentPage * resultsPerPage, totalResults );
+		return `${ start } – ${ end }`;
+	};
+
+	// Block properties and classes.
+	const blockProps = useBlockProps( {
+		className: clsx( className, 'query-total-block' ),
+	} );
```

Comment:
Why do we have the additional `query-total-block` class here? Can we not just use the `wp-block-query-total` class we already get out of the box? 

```suggestion
	const blockProps = useBlockProps();
```

Reply:
umm apologies 🙌🏻, I just missed that part.